### PR TITLE
PMM-7: restore github actions and fix tests

### PR DESCRIPTION
**Pull request by ritbl**
Most of the code has been restored from [main_old](https://github.com/percona/mysqld_exporter/commits/main_old) branch.
Fixed tests and DB image configuration, so that it is available to host.

### CODE REVIEW

File: test-db-init/init.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+GRANT ALL ON *.* to user@'%' IDENTIFIED BY 'password';
```
Comment (ritbl) [inline test-db-init/init.sql:1]: newer mysql instances do not allow to connect from non-local hosts;
mysql container will be running over NAT, not allowing to connect to it from host,;
this command will allow to connect from any host.

File: mysqld_exporter_test.go [hunk 2]
```diff
@@ -227,7 +227,9 @@ func testLandingPage(t *testing.T, data bin) {
 		data.path,
 		"--web.listen-address", fmt.Sprintf(":%d", data.port),
 	)
-	cmd.Env = append(os.Environ(), "DATA_SOURCE_NAME=127.0.0.1:3306")
+	cmd.Env = append(os.Environ(), "DATA_SOURCE_NAME=tcp:(127.0.0.1:3306)/")
```
Comment (ritbl) [inline mysqld_exporter_test.go:230]: fix DSN

File: mysqld_exporter_test.go [hunk 3]
```diff
@@ -227,7 +227,9 @@ func testLandingPage(t *testing.T, data bin) {
...
+	cmd.Stdout = os.Stdout
```
Comment (ritbl) [inline mysqld_exporter_test.go:231]: bind STDOUT and STDERR so that we can see that process has failed

File: mysqld_exporter_test.go [hunk 4]
```diff
@@ -245,8 +247,16 @@ func testLandingPage(t *testing.T, data bin) {
 	expected := `<html>
 <head><title>MySQLd exporter</title></head>
 <body>
-<h1>MySQLd exporter</h1>
-<p><a href='/metrics'>Metrics</a></p>
+<h1>MySQL 3-in-1 exporter</h1>
```
Comment (ritbl) [inline mysqld_exporter_test.go:250]: fix test

File: .github/workflows/go.yml [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+  pull_request:
+
+jobs:
+  test:
+    name: Test
+    strategy:
+      matrix:
+        go-version:
+          - 1.17
+        mysql-image:
```
Comment (ritbl) [inline .github/workflows/go.yml:18]: disabled mysql/maria 5.5, as they are not supported

File: collector/exporter_test.go [hunk 6]
```diff
@@ -89,6 +89,6 @@ func TestGetMySQLVersion(t *testing.T) {
 		convey.So(err, convey.ShouldBeNil)
 		defer db.Close()
 
-		convey.So(getMySQLVersion(db, logger), convey.ShouldBeBetweenOrEqual, 5.6, 11.0)
```
Comment (ritbl) [inline collector/exporter_test.go:92]: we run tests against 5.5

File: tools/go.mod [hunk 7]
```diff
@@ -0,0 +1,193 @@
+module github.com/percona/mysqld_exporter/tools
```
Comment (ritbl) [inline tools/go.mod:1]: all /tools content has been copied from old code, without modification

**Review by atymchuk — APPROVED** [review]

**Review by JiriCtvrtka — APPROVED** [review]

File: Makefile [hunk 8]
```diff
@@ -11,26 +11,80 @@
...
+	@echo ">> running short tests"
+	@$(GO) test -short -race $(pkgs)
 
-export PMM_RELEASE_PATH?=.
+test:             ## Run all tests
+	@echo ">> running tests"
+	@$(GO) test -race $(pkgs)
 
-release:
-	go build -ldflags="$(GO_BUILD_LDFLAGS)" -o $(PMM_RELEASE_PATH)/mysqld_exporter
```
Comment (BupycHuk) [inline Makefile:36]: PMM FB build is broken, because `make release` in `mysqld_exporter` was removed.

File: collector/exporter_test.go (hunk 6 above)
Comment (BupycHuk) [reply collector/exporter_test.go:92]: any reason for that?

**Pull request merged**